db/hints/manager.cc

```diff
@@ -56,6 +56,13 @@ const std::string manager::FILENAME_PREFIX("HintsLog" + commitlog::descriptor::S
 const std::chrono::seconds manager::hint_file_write_timeout = std::chrono::seconds(2);
 const std::chrono::seconds manager::hints_flush_period = std::chrono::seconds(10);
 
+const std::chrono::seconds manager::end_point_hints_manager::sender::concurrency_limiter::update_interval = std::chrono::seconds(1);
+const std::chrono::seconds manager::end_point_hints_manager::sender::concurrency_limiter::update_delay_after_timeout = std::chrono::seconds(10);
+
+thread_local std::uniform_int_distribution<uint64_t> manager::end_point_hints_manager::sender::concurrency_limiter::_next_delay_distribution{1, 120};
+
+static thread_local std::mt19937_64 rand_gen(std::random_device{}());
```

Comment:
Here, thread_local makes much more sense, since the generator state is huge. Still, consider making it a (private) member of the manager - we have 1 manager per shard anyway, and we drop the tls guarding code.

Reply:
Actually, we have two managers per shard: one for regular hints, and the other one for regular hints - the generator is shared between them. The generator is seeded by random_device, so the number of instantiations should be minimized. Similarly to the `_next_delay_distribution`, the generator will be used relatively rarely (once per second for each active hints manager).